### PR TITLE
Prepare for fix to https://github.com/flutter/flutter/issues/109339.

### PR DESCRIPTION
**Pull request by stereotype441**
Currently, in some circumstances where a subclasses of
`PlatformInterface` erroneously uses `implements` rather than
`extends`, a `NoSuchMethodError` will be thrown (in spite of the
documentation at
https://pub.dev/documentation/plugin_platform_interface/latest/plugin_platform_interface/PlatformInterface/verify.html
claiming that `AssertionError` will be thrown).

After https://github.com/flutter/flutter/issues/109339 is fixed, the
correct type of exception will be thrown.  To avoid tests breakages
when the fix happens, we need to modify these tests so that they don't
care what kind of exception is thrown.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [ ] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
> This change should also be exempt from a version bump

The heuristics know that here since it's test-only, so we only need to override the changelog check.

File: packages/in_app_purchase/in_app_purchase_platform_interface/test/in_app_purchase_platform_test.dart [hunk 1]
```diff
@@ -128,7 +135,14 @@ void main() {
...
       expect(InAppPurchasePlatformAddition.instance, isNull);
-    });
+      // In versions of `package:plugin_platform_interface` prior to fixing
+      // https://github.com/flutter/flutter/issues/109339, an attempt to
+      // implement a platform interface using `implements` would sometimes throw
+      // a `NoSuchMethodError` and other times throw an `AssertionError`.  After
+      // the issue is fixed, an `AssertionError` will always be thrown.  For the
+      // purpose of this test, we don't really care what exception is thrown, so
+      // just allow any exception.
+    }, throwsA(anything));
```
Comment (stuartmorgan) [inline packages/in_app_purchase/in_app_purchase_platform_interface/test/in_app_purchase_platform_test.dart]: This doesn't compile, which is why everything is red.

Also this test seems to just be mis-named, since it's just checking that the default is null, not trying to use implements. And implements is valid: https://github.com/flutter/plugins/blob/7686be7ecbe1cc049423715b5e2d897cb63a43c1/packages/in_app_purchase/in_app_purchase_platform_interface/lib/src/in_app_purchase_platform_addition.dart#L11-L13 (and the test below) so you can just revert this (and ideally fix the name).

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/in_app_purchase/in_app_purchase_platform_interface/test/in_app_purchase_platform_test.dart (hunk 1 above)
Comment (stuartmorgan) [reply packages/in_app_purchase/in_app_purchase_platform_interface/test/in_app_purchase_platform_test.dart]: Oops, midair collision.

Could you fix the name though, now that we've noticed it, so it doesn't confuse anyone else the way it did you?

Comment (stereotype441) [reply packages/in_app_purchase/in_app_purchase_platform_interface/test/in_app_purchase_platform_test.dart]: No problem.  Done!

**Review by stuartmorgan — APPROVED** [review]
LGTM, thanks!
